sklearn/preprocessing/tests/test_data.py

```diff
@@ -1500,13 +1506,37 @@ def test_one_hot_encoder_sparse():
     # test exception on wrong init param
     assert_raises(TypeError, OneHotEncoder(n_values=np.int).fit, X)
 
+
+def test_one_hot_encoder_error_on_negative():
+    # Negative numerical values in inputs should raise an exception
```

Comment:
Okay. Do we have a test that checks negative integers are fine in the `auto-strict` case?